pyhik/hikvision.py

```diff
@@ -132,6 +134,72 @@ def current_event_states(self):
         """Return Event states dictionary"""
         return self.event_states
 
+    def get_motion_detection(self):
+        url = '%s/MotionDetection/1' % self.root_url
+        try:
+            response = self.hik_request.get(url)
+        except requests.exceptions.RequestException as err:
+            _LOGGING.error('Unable to fetch MotionDetection, error: %s', err)
+            return None
+
+        if response.status_code == requests.codes.unauthorized:
+            _LOGGING.error('Authentication failed')
+            return None
+
+        if response.status_code != requests.codes.ok:
+            # If we didn't receive 200, abort
+            _LOGGING.debug('Unable to fetch motion detection.')
+            return None
+
+        try:
+            tree = ET.fromstring(response.text)
+            nmsp = tree.tag.split('}')[0].strip('{')
```

Comment:
You shouldn't need to check the namespace a second time here.  We fetch it during the device info call in the main initialization function.

Reply:
I see that it does indeed change with the URL path you've choosen, but it remains consistent when using the one defined in the API docs that I had saved as a comment:
http://X.X.X.X/ISAPI/System/Video/inputs/channels/1/motionDetection

Is there a reason why you choose a different endpoint?

Reply:
That endpoint returns all the scheduling data together with the state of motion detection. The response on the one I used contains very little additional data, and I thought it was unnecessary to be sending/retrieving all that data if there's no need.

Provided that the XML schema indeed changes between different calls, I consider it more safe to always check it and set it accordingly.